utils/utils.go

```diff
@@ -29,16 +30,28 @@ var (
 	headerRegexp = regexp.MustCompile(`\.(h|hpp|inc)$`)
 )
 
+var compileSrcsExt = []string{".c", ".s", ".cpp", ".cc", ".S"}
```

Comment:
Fix: Isn't this unused?

Reply:
Ah sorry, this is used.

Can we implement this by matching the pattern of `IsHeader`/`IsNotHeader`? So make this a regex like `headerRegexp`?

Reply:
Sure changed to regex